Fresh prod install through docker fails with npm ERR! Failed to access /var/www/.cache/Cypress

# Bug report

### Actual Behavior

Following the docker readme, the test install succeeds but the prod fails. 

### CDash Version

v3.2.1

### Additional Information

Fresh Ubuntu Jammy on a KVM, docker is installed through the official Docker repository.
I follow the readme to the letter, except for the APP_KEY where I copy the one given after the first attempt fails.
Here's the log of the error on the second attempt:

```
[+] Running 2/0
 ✔ Container cdash_postgres  Created                                                                                                                                                                                                                                        0.0s 
 ✔ Container cdash           Created                                                                                                                                                                                                                                        0.0s 
Attaching to cdash, cdash_postgres, cdash_worker
cdash_postgres  | 
cdash_postgres  | PostgreSQL Database directory appears to contain a database; Skipping initialization
cdash_postgres  | 
cdash_postgres  | 2024-01-15 10:35:19.020 UTC [1] LOG:  starting PostgreSQL 16.1 (Debian 16.1-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
cdash_postgres  | 2024-01-15 10:35:19.020 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
cdash_postgres  | 2024-01-15 10:35:19.020 UTC [1] LOG:  listening on IPv6 address "::", port 5432
cdash_postgres  | 2024-01-15 10:35:19.023 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
cdash_postgres  | 2024-01-15 10:35:19.027 UTC [29] LOG:  database system was shut down at 2024-01-15 10:32:55 UTC
cdash_postgres  | 2024-01-15 10:35:19.038 UTC [1] LOG:  database system is ready to accept connections
cdash           | =================================================================================
cdash           | Beginning CDash installation...
cdash           | Enabling maintenance mode...
cdash           | 
cdash           |    INFO  Application is already down.  
cdash           | 
cdash           | Updating vendor dependencies...
cdash           | npm WARN deprecated har-validator@5.1.5: this library is no longer supported
cdash           | npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
cdash           | npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
cdash           | npm WARN deprecated protractor@7.0.0: We have news to share - Protractor is deprecated and will reach end-of-life by Summer 2023. To learn more and find out about other options please refer to this post on the Angular blog. Thank you for using and contributing to Protractor. https://goo.gle/state-of-e2e-in-angular
cdash           | npm ERR! code 1
cdash           | npm ERR! path /cdash/node_modules/cypress
cdash           | npm ERR! command failed
cdash           | npm ERR! command sh -c node index.js --exec install
cdash           | npm ERR! Cypress cannot write to the cache directory due to file permissions
cdash           | npm ERR! 
cdash           | npm ERR! See discussion and possible solutions at
cdash           | npm ERR! https://github.com/cypress-io/cypress/issues/1281
cdash           | npm ERR! 
cdash           | npm ERR! ----------
cdash           | npm ERR! 
cdash           | npm ERR! Failed to access /var/www/.cache/Cypress:
cdash           | npm ERR! 
cdash           | npm ERR! EACCES: permission denied, mkdir '/var/www/.cache'
cdash           | npm ERR! 
cdash           | npm ERR! ----------
cdash           | npm ERR! 
cdash           | npm ERR! Platform: linux-x64 (Debian - 12.4)
cdash           | npm ERR! Cypress Version: 13.6.2
cdash           | 
cdash           | npm ERR! A complete log of this run can be found in: /var/www/.npm/_logs/2024-01-15T10_35_21_402Z-debug-0.log
cdash           | 
cdash           | * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
cdash           | *                            INSTALLATION FAILURE!                            *
cdash           | *                                                                             *
cdash           | * An error occurred while installing CDash and CDash was not able to recover  *
cdash           | * automatically.  If you believe that this is a problem with CDash, please    *
cdash           | * report the error here:                                                      *
cdash           | *                                                                             *
cdash           | * -> https://github.com/Kitware/CDash/issues/new                              *
cdash           | *                                                                             *
cdash           | * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
cdash           | 
dependency failed to start: container cdash exited (1)
cdash exited with code 1
```

I haven't been able to find any way around this...
Any help would be most welcome. 
Cheers

--- a/docker/docker-entrypoint.sh
+++ b/docker/docker-entrypoint.sh
@@ -6,7 +6,11 @@ php artisan key:check || exit 1
 
 # If the "start-website" argument was provided, start the web server
 if [ "$1" = "start-website" ] ; then
-  bash /cdash/install.sh
+  if [ "$DEVELOPMENT_BUILD" = "1" ]; then
+    bash /cdash/install.sh --dev
+  else
+    bash /cdash/install.sh
+  fi
 
   echo "Starting Apache..."
 

--- a/install.sh
+++ b/install.sh
@@ -1,7 +1,17 @@
 #!/bin/bash
 
+if [[ "$1" == "--dev" ]]; then
+  DEVELOPMENT=true
+else
+  DEVELOPMENT=false
+fi
+
 echo "=================================================================================";
-echo "Beginning CDash installation...";
+if $DEVELOPMENT; then
+  echo "Beginning development CDash installation..."
+else
+  echo "Configuring production CDash installation..."
+fi
 
 error_handler() {
 echo "
@@ -29,11 +39,16 @@ echo "Enabling maintenance mode..."
 php artisan down --render="maintenance" --refresh=5
 
 echo "Updating vendor dependencies..."
-npm install
-composer install
+if $DEVELOPMENT; then
+  npm install
+  composer install
+else
+  npm install --omit=dev
+  composer install --no-dev --optimize-autoloader
+fi
 
 echo "Running migrations..."
-php artisan migrate
+php artisan migrate --force
 php artisan version:set
 
 echo "Clearing caches..."

--- a/package.json
+++ b/package.json
@@ -50,7 +50,8 @@
     "vue": "^3.3.8",
     "vue-loader": "^17.4.2",
     "vuedraggable": "^4.1.0",
-    "webpack": "^5.89.0"
+    "webpack": "^5.89.0",
+    "resolve-url-loader": "^5.0.0"
   },
   "devDependencies": {
     "@vue/test-utils": "^2.4.3",
@@ -62,7 +63,6 @@
     "jest": "^29.7.0",
     "jest-environment-jsdom": "^29.7.0",
     "postcss": "^8.4.32",
-    "protractor": "^7.0.0",
-    "resolve-url-loader": "^5.0.0"
+    "protractor": "^7.0.0"
   }
 }

